shm/Map.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <vector>
+#include "island.hpp"
+#include <random>
+#include <algorithm>
+#include <utility>
+#include <iostream>
+#include "coordinates.hpp"
```

Comment:
Sort headers. Include form standard libraries than from the project and also keep it lexicographically order.